src/extension.js

```diff
@@ -410,17 +319,55 @@ const ServiceIndicator = GObject.registerClass({
         this.settings.run_dispose();
 
         // Destroy the PanelMenu.SystemIndicator actors
-        this._item.destroy();
         this.menu.destroy();
 
-        delete AggregateMenu._gsconnect;
+        delete QuickSettingsMenu._gsconnect;
         super.destroy();
     }
 });
 
+const FeatureIndicator = GObject.registerClass(
+class FeatureIndicator extends QuickSettings.SystemIndicator {
+    _init() {
+        super._init();
 
-var serviceIndicator = null;
+        // GSettings
+        this.settings = new Gio.Settings({
+            settings_schema: Config.GSCHEMA.lookup(
+                'org.gnome.Shell.Extensions.GSConnect',
+                null
+            ),
+            path: '/org/gnome/shell/extensions/gsconnect/',
+        });
+
+        // Create the icon for the indicator
+        this._indicator = this._addIndicator();
+        this._indicator.icon_name = 'org.gnome.Shell.Extensions.GSConnect-symbolic';
+        // Hide the indicator by default
+        this._indicator.visible = false;
+
+        // Create the toggle menu and associate it with the indicator
+        this.quickSettingsItems.push(new ServiceIndicator());
+
+        // Add the indicator to the panel and the toggle to the menu
+        QuickSettingsMenu._indicators.insert_child_at_index(this, 0);
+        QuickSettingsMenu._addItems(this.quickSettingsItems);
+    }
+
+    destroy() {
+        // Set enabled state to false to kill the service on destroy
+        this.settings.set_boolean('enabled', false);
+        this.quickSettingsItems.forEach(item => item.destroy());
+        // Destroy indicator
+        // Workaround: Disabling and enabling extension multiple times increases
+        // padding between new indicator and other indicators. Hide indicator before
+        // destroying to workaround this bug.
+        this._indicator.visible = false;
+        this._indicator.destroy();
+    }
+});
 
+var featureIndicator = null;
```

Comment:
```suggestion
var serviceIndicator = null;
```

Renaming for consistency (ServiceIndicator, ServiceToggle).